src/communication/rpc_impl.rs

```diff
@@ -0,0 +1,486 @@
+/********************************************************************************
```

Comment:
What do you think the chances are that we'd have > 1 RPC implementation in the future in up-rust?

Worthwhile to name this file something more specific s.t. in the future we can have `rpc_in_memory_impl.rs`, `rpc_db_backed_impl.rs`, and so on?

Or just leave this and we'll navigate that as it comes up?

Reply:
Good point, will rename the file :+1: